### PR TITLE
Compatibility with ffi rzmq

### PR DESCRIPTION
**Pull request by domitry**
This pull-req is only for being checked by @minad . Test suites will soon be added.

Copied from the message in the last commit:

> I don't know what is the best solution about add_runtime_dependency.
> If this line is not removed Windows users will so confused, but if not
> users will wonder how to solve LoadError 'ffi-rzmq' or 'rbczmq'

Anyway we can use IRuby on Windows again with this pull-req and #49 !

### CODE REVIEW

File: iruby.gemspec [hunk 1]
```diff
@@ -28,7 +28,6 @@ Gem::Specification.new do |s|
   s.add_development_dependency 'minitest', '~> 5.6'
 
   s.add_runtime_dependency 'bond', '~> 0.5'
-  s.add_runtime_dependency 'rbczmq', '~> 1.7'
```
Comment (minad) [inline iruby.gemspec]: Since we removed it from the dependencies - we have to print a message if the loading of either one fails.

**Comment by minad** [comment]
It seems that you monkey patch the session class (or overwrite it). It would be better to require rbczmq etc in the appropriate session files and create two separate session files for the backends.

something like this
~~~  ruby
begin
  require 'session/rbczmq'
rescue LoadError
  begin
    require 'session/ffi'
  rescue LoadError
     STDERR.puts "you have to install..."
  end
end
~~~

**Comment by domitry** [comment]
OK, I divided IRuby::Session into two files and add a message displayed when both `require 'rbczmq'` and `require 'ffi-rzmq'` fails. 

Now two methods common between two files are in lib/iruby/session/base.
I think they can be moved to each Session when someone need to modify them.

**Comment by minad** [comment]
Ok looks better. However for some stylistic reason I prefer to have a mixin instead of this SessionBase.
Otherwise it seems perfect :+1:

**Comment by minad** [comment]
Why is the serialize routine different? Could this not also moved to the mixin SessionSerialize or SessionBase?

**Comment by domitry** [comment]
> Why is the serialize routine different?

It reflects that I tried not to make any changes to the code of rbczmq version before.
Fixed.

**Comment by minad** [comment]
thx great work!

**Comment by domitry** [comment]
@minad BTW can you release the newest version including patches related to ffi-rzmq compativility?
I heard that Mac users still fail to install iruby from rubygems because of rbczmq.

**Comment by minad** [comment]
Is it ok if I add you as a gem owner? :smile:

**Comment by domitry** [comment]
Yeah I'm so glad if you allow!! :relaxed:

**Comment by domitry** [comment]
confirmed. Thank you!!

**Comment by minad** [comment]
great thx!

**Pull request merged**